gemini-cli: native keytar module fails to load on macOS

### `brew config` AND `brew doctor` output OR `brew gist-logs <formula>` link

```shell
HOMEBREW_VERSION: 5.1.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 299ab19496df6fc76e8356c0b87100ff91ef827e
Last commit: 27 hours ago
Branch: stable
Core tap JSON: 13 Apr 23:07 UTC
Core cask tap JSON: 13 Apr 23:07 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DOWNLOAD_CONCURRENCY: 20
HOMEBREW_FORBID_PACKAGES_FROM_PATHS: set
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 4.0.2 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/4.0.2_1/bin/ruby
CPU: deca-core 64-bit arm_donan
Clang: 21.0.0 build 2100
Git: 2.50.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 26.4.1-arm64
CLT: 26.4.0.0.1774242506
Xcode: 26.4
Metal Toolchain: N/A
Rosetta 2: false
```

### Verification

- [x] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [x] I ran `brew update` and am still able to reproduce my issue.
- [x] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [x] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.
- [x] My issue is not about a failure to build a formula from source.

### What were you trying to do (and why)?

Use gemini-cli installed via Homebrew.

### What happened (include all command output)?

When running `gemini` commands, they run, but they also include this warning:
```
Keychain initialization encountered an error: Cannot find module '../build/Release/keytar.node'
Require stack:
- /opt/homebrew/Cellar/gemini-cli/0.37.1/libexec/lib/node_modules/@google/gemini-cli/node_modules/keytar/lib/keytar.js
Using FileKeychain fallback for secure storage.
Loaded cached credentials.
```

### What did you expect to happen?

That warning message shouldn't be there. The gemini-cli installed via Homebrew should correctly compile and link the @github/keytar native module during installation, enabling native macOS Keychain support and eliminating the fallback error.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. `brew install gemini-cli`
2. `gemini --list-sessions`

Result:

No previous sessions found for this project.Keychain initialization encountered an error: Cannot find module '../build/Release/keytar.node'
Require stack:
- /opt/homebrew/Cellar/gemini-cli/0.37.1/libexec/lib/node_modules/@google/gemini-cli/node_modules/keytar/lib/keytar.js
Using FileKeychain fallback for secure storage.
Loaded cached credentials.
```

> ### `brew config` AND `brew doctor` output OR `brew gist-logs <formula>` link
> HOMEBREW_VERSION: 5.1.6
> ORIGIN: https://github.com/Homebrew/brew
> HEAD: 299ab19496df6fc76e8356c0b87100ff91ef827e
> Last commit: 27 hours ago
> Branch: stable
> Core tap JSON: 13 Apr 23:07 UTC
> Core cask tap JSON: 13 Apr 23:07 UTC
> HOMEBREW_PREFIX: /opt/homebrew
> HOMEBREW_CASK_OPTS: []
> HOMEBREW_DOWNLOAD_CONCURRENCY: 20
> HOMEBREW_FORBID_PACKAGES_FROM_PATHS: set
> HOMEBREW_MAKE_JOBS: 10
> Homebrew Ruby: 4.0.2 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/4.0.2_1/bin/ruby
> CPU: deca-core 64-bit arm_donan
> Clang: 21.0.0 build 2100
> Git: 2.50.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
> Curl: 8.7.1 => /usr/bin/curl
> macOS: 26.4.1-arm64
> CLT: 26.4.0.0.1774242506
> Xcode: 26.4
> Metal Toolchain: N/A
> Rosetta 2: false
> ### Verification
> * [x]  My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.[x]  I ran `brew update` and am still able to reproduce my issue.[x]  I have resolved all warnings from `brew doctor` and that did not fix my problem.[x]  I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.[x]  My issue is not about a failure to build a formula from source.
> 
> ### What were you trying to do (and why)?
> Use gemini-cli installed via Homebrew.
> 
> ### What happened (include all command output)?
> When running `gemini` commands, they run, but they also include this warning:
> 
> ```
> Keychain initialization encountered an error: Cannot find module '../build/Release/keytar.node'
> Require stack:
> - /opt/homebrew/Cellar/gemini-cli/0.37.1/libexec/lib/node_modules/@google/gemini-cli/node_modules/keytar/lib/keytar.js
> Using FileKeychain fallback for secure storage.
> Loaded cached credentials.
> ```
> 
> ### What did you expect to happen?
> That warning message shouldn't be there. The gemini-cli installed via Homebrew should correctly compile and link the @github/keytar native module during installation, enabling native macOS Keychain support and eliminating the fallback error.
> 
> ### Step-by-step reproduction instructions (by running `brew` commands)
> 1. `brew install gemini-cli`
> 2. `gemini --list-sessions`
> 
> Result:
> 
> No previous sessions found for this project.Keychain initialization encountered an error: Cannot find module '../build/Release/keytar.node'
> Require stack:
> - /opt/homebrew/Cellar/gemini-cli/0.37.1/libexec/lib/node_modules/@google/gemini-cli/node_modules/keytar/lib/keytar.js
> Using FileKeychain fallback for secure storage.
> Loaded cached credentials.

--- a/Formula/g/gemini-cli.rb
+++ b/Formula/g/gemini-cli.rb
@@ -6,12 +6,12 @@ class GeminiCli < Formula
   license "Apache-2.0"
 
   bottle do
-    sha256 cellar: :any_skip_relocation, arm64_tahoe:   "469d65f595b507baef8cf349a363f2ef3844d0d89d805ebdf146707620616569"
-    sha256 cellar: :any_skip_relocation, arm64_sequoia: "469d65f595b507baef8cf349a363f2ef3844d0d89d805ebdf146707620616569"
-    sha256 cellar: :any_skip_relocation, arm64_sonoma:  "469d65f595b507baef8cf349a363f2ef3844d0d89d805ebdf146707620616569"
-    sha256 cellar: :any_skip_relocation, sonoma:        "07096f835e06edf9ccc591603c5a9f0ce7a3ae877eb6bc36ce79e9e007b6b922"
-    sha256 cellar: :any_skip_relocation, arm64_linux:   "8fcb15ae80ffe151cc65ba8f2fc92c34bc0e88fadef7d0c55c5cd88cbc063ca2"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:  "b351b6da6e4e35b234436c4cabb863f8a220d696ad85b19ee72c02acb61d4817"
+    sha256                               arm64_tahoe:   "5c467a352f626de109e4a5a1f158a1fae4da13df2d91b80fa6a3958a4069fa19"
+    sha256                               arm64_sequoia: "e0d38b7fc0bb41fc8ee9eeba6627cb2ebcfa7bf8b75bc5a2cb41322300888a5d"
+    sha256                               arm64_sonoma:  "945de33839b513a8ecbdc789f7b98422920314d6a7e50872e4fe814dac224bf0"
+    sha256                               sonoma:        "d1d4b2c29a4b91b943281c39cf8d18899a6767aa87875e895f86f84bcebb0f6c"
+    sha256 cellar: :any_skip_relocation, arm64_linux:   "9b6f8a16bda9f5b9431df4df52d09f3202ca5e8433e7cf167655bf2c34ce7386"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:  "e1f827611e2619dce24e86efe7a85dda58174c50255f115a11ddbba5a6a35317"
   end
 
   depends_on "node"

--- a/Formula/g/gemini-cli.rb
+++ b/Formula/g/gemini-cli.rb
@@ -1,8 +1,8 @@
 class GeminiCli < Formula
   desc "Interact with Google Gemini AI models from the command-line"
   homepage "https://github.com/google-gemini/gemini-cli"
-  url "https://registry.npmjs.org/@google/gemini-cli/-/gemini-cli-0.38.2.tgz"
-  sha256 "9b0c752cfe9375370e1812f37afffd97387b99df71e64cea53e588a25f4d688c"
+  url "https://registry.npmjs.org/@google/gemini-cli/-/gemini-cli-0.39.0.tgz"
+  sha256 "8b0093325fd6a316fa273b2a7b9c98ff0d714b0560ae1fd48363b003c993d0c8"
   license "Apache-2.0"
 
   bottle do
@@ -16,6 +16,12 @@ class GeminiCli < Formula
 
   depends_on "node"
 
+  on_linux do
+    depends_on "pkgconf" => :build
+    depends_on "glib"
+    depends_on "libsecret"
+  end
+
   def install
     system "npm", "install", *std_npm_args
     bin.install_symlink libexec.glob("bin/*")
@@ -26,6 +32,11 @@ class GeminiCli < Formula
     node_modules = libexec/"lib/node_modules/@google/gemini-cli/node_modules"
     node_modules.glob("{bare-fs,bare-os,bare-url,tree-sitter-bash,node-pty}/prebuilds/*").each do |dir|
       rm_r(dir) if dir.basename.to_s != "#{os}-#{arch}"
+    end
+
+    cd node_modules/"@github/keytar" do
+      rm_r "prebuilds"
+      system "npm", "run", "build"
     end
   end
 

